Tests/CryptoTests/Authenticated Encryption/AES-GCM-Runner.swift

```diff
@@ -155,7 +155,7 @@ class AESGCMTests: XCTestCase {
             let ciphertext = try! AES.GCM.seal(message, using: key, nonce: nonce, authenticating: aad)
             let recoveredPlaintext = try! AES.GCM.open(ciphertext, using: key, authenticating: aad)
 
-            XCTAssert(Array(recoveredPlaintext) == Array(message))
+            XCTAssertEqual(Array(recoveredPlaintext), Array(message))
```

Comment:
Not relating to your changes, but while making improvments: We should make use of `line` argument of `XCTAssertFOO` functions when using *nested calls* within tests.

So the test `testRoundTripDataProtocols` which makes use of nested function `roundTrip` can be improved from:

```swift
 func testRoundTripDataProtocols() throws {
        func roundTrip<Message: DataProtocol, AAD: DataProtocol>(message: Message, aad: AAD) {
            let key = SymmetricKey(size: .bits256)
            let nonce = AES.GCM.Nonce()

            let ciphertext = try! AES.GCM.seal(message, using: key, nonce: nonce, authenticating: aad)
            let recoveredPlaintext = try! AES.GCM.open(ciphertext, using: key, authenticating: aad)

            XCTAssertEqual(Array(recoveredPlaintext), Array(message))
        }

        let message = Array("Hello, world, it's AES-GCM!".utf8)
        let aad = Array("I heard you like Counter Mode, so I put a Galois on it".utf8)
        let (contiguousMessage, discontiguousMessage) = message.asDataProtocols()
        let (contiguousAad, discontiguousAad) = aad.asDataProtocols()

        roundTrip(message: contiguousMessage, aad: contiguousAad)
        roundTrip(message: discontiguousMessage, aad: contiguousAad)
        roundTrip(message: contiguousMessage, aad: discontiguousAad)
        roundTrip(message: discontiguousMessage, aad: discontiguousAad)
    }
```

to:

```swift
   func testRoundTripDataProtocols() throws {
        func roundTrip<Message: DataProtocol, AAD: DataProtocol>(message: Message, aad: AAD, line: UInt = #line) {
            let key = SymmetricKey(size: .bits256)
            let nonce = AES.GCM.Nonce()

            let ciphertext = try! AES.GCM.seal(message, using: key, nonce: nonce, authenticating: aad)
            let recoveredPlaintext = try! AES.GCM.open(ciphertext, using: key, authenticating: aad)

            XCTAssertEqual(Array(recoveredPlaintext), Array(message), line: line)
        }

        let message = Array("Hello, world, it's AES-GCM!".utf8)
        let aad = Array("I heard you like Counter Mode, so I put a Galois on it".utf8)
        let (contiguousMessage, discontiguousMessage) = message.asDataProtocols()
        let (contiguousAad, discontiguousAad) = aad.asDataProtocols()

        roundTrip(message: contiguousMessage, aad: contiguousAad)
        roundTrip(message: discontiguousMessage, aad: contiguousAad)
        roundTrip(message: contiguousMessage, aad: discontiguousAad)
        roundTrip(message: discontiguousMessage, aad: discontiguousAad)
    }
```

Observe the appending of `, line: line)` at end of `XCTAssertEqual`, this ensures that failing test marker in Xcode propages to the call site of each `roundTrip` call (lines: 166-169)

Reply:
I have not checked other nested function calls in tests yet, just found this now. If calling a helper function (from a `test***` function) that are in a different file one should pass `file: file` as well as `line: line`.

Reply:
ooh great call. This is [right up my alley](https://github.com/realm/SwiftLint/pull/3040) 😁

Reply:
Swifty minds think a like